### ISSUE TITLE
$pageimage->maxSize() can produce unexpected results

### ISSUE THREAD

**Issue by Toutouwai** [processwire/processwire-issues#454]
### Short description of the issue

$pageimage->maxSize() can produce unexpected results.

Example:

````php
// for $image sized 1200x800
$resized = $image->maxSize(600, 300);
// $resized is 600x400, but should be 450x300
````

### Suggestion for a possible fix

````php
public function maxSize($width, $height, $options = array()) {
    $options['upscaling'] = false;
    $options['cropping'] = false;
    return $this->size($width, $height, $options);
}
````

...which is effectively what @BernhardBaumrock suggested [here](https://github.com/processwire/processwire-issues/issues/57#issuecomment-307063197).

Only downside to this approach is that the variation in the example above is named with a '600x300' suffix despite actually being 450x300, so it would result in the wrong image being served if a 600x300 variation was actually requested at some point. But this is something that should probably be resolved separately within the size() method (i.e. ensure the variation suffix is consistent with the dimensions of the variation produced).

### Setup/Environment

* ProcessWire version: 3.0.85

**Comment by ryancramerdesign** [comment]
Thanks, I've pushed your suggested fix. Regarding the filename, the intention is that it represents the request rather than the result, as the filenames are basically a type of cache to translate a request to a file, without having to run through a bunch of calculations first. But that's a good point about how it could get confused with a regular size() image. I'm thinking that would be a pretty rare occurrence, but if we can avoid it, we should. I've updated it so that it uses a suffix on images generated by maxSize(), but only for new installations from this date forward, so as not to force re-creation of images on existing installations.
